lib/charms/observability_libs/v0/kubernetes_service_patch.py

```diff
@@ -157,22 +168,32 @@ def _service_object(
                 application name will be used.
             service_type: desired type of K8s service. Default value is in line with ServiceSpec's
                 default value.
+            additional_labels: Labels to be added to the kubernetes service (by default only
+                "app.kubernetes.io/name" is set to the service name)
+            additional_selectors: Selectors to be added to the kubernetes service (by default only
+                "app.kubernetes.io/name" is set to the service name)
 
         Returns:
             Service: A valid representation of a Kubernetes Service with the correct ports.
         """
         if not service_name:
             service_name = self._app
+        labels = {"app.kubernetes.io/name": service_name}
```

Comment:
This is not correct, and is potentially bad.

A merge patch is _probably_ ok, but the risk of `_delete_and_create_service` clobbering this is bad.

We should fetch the actual labels and selectors before blindly trying to update them. Juju **does** set values for these, and they may change out from underneath us. Even the default values include more than `app.kubernetes.io/name: foo`

```yaml
apiVersion: v1
kind: Service
metadata:
  labels:
    app.juju.is/created-by: prometheus
    app.kubernetes.io/managed-by: juju
    app.kubernetes.io/name: prometheus
  selector:
    app.kubernetes.io/name: prometheus
```

We'd have to instantiate the `Client` object first and actually query it, but the safety is worthwhile

Reply:
In the case where the user provides a `service_name`, the `_delete_and_create_service` first `gets` a service before deleting it, then recreates the same service with a different name. This new service will then those same juju specific labels/selectors as before. Finally the service is merged patched with the new user provided labels.

Reply:
That's reassuring. I didn't look into the exact return from `client.Get`